File: Java/src/main/java/br/com/locaweb/locamail/api/repository/AgendaConvidadoRepository.java
package br.com.locaweb.locamail.api.repository;

import br.com.locaweb.locamail.api.model.AgendaConvidado;
import br.com.locaweb.locamail.api.model.Pasta;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.List;


@Repository
public interface AgendaConvidadoRepository extends JpaRepository<AgendaConvidado, Long> {

    @Query(value = "SELECT id_convidado FROM T_LCW_AGENDA_CONVIDADO WHERE id_agenda = :id_agenda", nativeQuery = true)
    public List<Long> listarIdConvidadoPorAgenda(@Param("id_agenda") Long id_agenda);

    @Query(value = "DELETE FROM T_LCW_AGENDA_CONVIDADO WHERE grupo_repeticao = :grupo_repeticao AND id_agenda != :id_agenda", nativeQuery = true)
    public void excluirPorGrupoRepeticaoExcetoIdAgenda(@Param("grupo_repeticao") Integer grupo_repeticao, @Param("id_agenda") Long id_agenda);

    @Query(value = "DELETE FROM T_LCW_AGENDA_CONVIDADO WHERE id_agenda = :id_agenda", nativeQuery = true)
    public void excluirPorIdAgenda(@Param("id_agenda") Long id_agenda);

    @Query(value = "DELETE FROM T_LCW_AGENDA_CONVIDADO WHERE id_agenda = :id_agenda AND id_convidado = :id_convidado", nativeQuery = true)
    public void excluirPorIdAgendaEIdConvidado(@Param("id_agenda") Long id_agenda, @Param("id_convidado") Long id_convidado);

}

File: Kotlin/app/src/main/java/br/com/fiap/locawebmailapp/database/repository/AgendaConvidadoRepository.kt
package br.com.fiap.locawebmailapp.database.repository

import android.content.Context
import br.com.fiap.locawebmailapp.database.dao.InstanceDatabase
import br.com.fiap.locawebmailapp.model.Agenda
import br.com.fiap.locawebmailapp.model.AgendaConvidado
import br.com.fiap.locawebmailapp.model.AgendaCor
import br.com.fiap.locawebmailapp.model.AgendaGrupoRepeticao
import br.com.fiap.locawebmailapp.model.Convidado
import br.com.fiap.locawebmailapp.model.IdConvidado

class AgendaConvidadoRepository(context: Context) {
    private val agendaConvidadoDao = InstanceDatabase.getDatabase(context).agendaConvidadoDao()

    fun criaAgendaConvidado(agendaConvidado: AgendaConvidado): Long {
        return agendaConvidadoDao.criarAgendaConvidado(agendaConvidado)
    }

    fun listarIdConvidadoPorAgenda(id_agenda: Long): List<IdConvidado> {
        return agendaConvidadoDao.listarIdConvidadoPorAgenda(id_agenda)
    }

    fun excluirPorGrupoRepeticaoExcetoIdAgenda(grupo_repeticao: Int, id_agenda: Long): Int {
        return agendaConvidadoDao.excluirPorGrupoRepeticaoExcetoIdAgenda(grupo_repeticao, id_agenda)
    }

    fun excluirPorIdAgenda(id_agenda: Long): Int {
        return agendaConvidadoDao.excluirPorIdAgenda(id_agenda)
    }

    fun excluirPorIdAgendaEIdConvidado(id_agenda: Long, id_convidado: Long): Int {
        return agendaConvidadoDao.excluirPorIdAgendaEIdConvidado(
            id_agenda = id_agenda,
            id_convidado = id_convidado
        )
    }

}